### PR TITLE
Email Verification

### PR DESCRIPTION
**Pull request by GAOChengzhan**
#### Register
image

#### Dialog
image

### CODE REVIEW

**Comment by GAOChengzhan** [comment]
The description in dialog and the button label may be discussed later to keep consistent across NDEx, Cy-web, Network Viewer, Iquery.

File: src/bootstrap.tsx [hunk 1]
```diff
@@ -43,14 +50,62 @@ loadingMessage.textContent = 'Initializing Cytoscape. Please wait...'
...
 
 const keycloak = new Keycloak(keycloakConfig)
+
+const handleVerify = async () => {
+  await keycloak.loadUserProfile()
+  window.location.reload()
+}
+
+const handleCancel = () => {
+  keycloak.logout()
```
Comment (d2fong) [inline src/bootstrap.tsx:60]: I think there is a parameter here called redirectUri that is used in this logout function.  Depending on how cytoscape web is deployed e.g. test.ndexbio.org/cytoscape or web.cytsocape.org we might need to send a argument here.

File: src/bootstrap.tsx [hunk 2]
```diff
@@ -43,14 +50,62 @@ loadingMessage.textContent = 'Initializing Cytoscape. Please wait...'
...
+}
+
+// Function to check if the user's email is verified
+const checkUserVerification = async () => {
+  try {
+    const ndexClient = new NDEx(appConfig.ndexBaseUrl)
+    ndexClient.setAuthToken(keycloak.token)
+    await ndexClient.getSignedInUser()
+    return {
+      isVerified: true,
```
Comment (d2fong) [inline src/bootstrap.tsx:70]: In my opinion, this function should always return a consistent object.  i.e. always return a object with `isVerified`, `userName` and `userEmail`, even if the the fields will not always be used by the calling function.

**Review by d2fong — COMMENTED** [review]
Looks good, we just need to see if we need we need to use a redirectUri when keycloak logs out across the various cyweb deployments.

**Review by d2fong — APPROVED** [review]

**Pull request merged**